tests/ui/derives/lack-of-perfect-derives.stderr

```diff
@@ -0,0 +1,240 @@
+error[E0369]: binary operation `==` cannot be applied to type `S<X, X, X>`
+  --> $DIR/lack-of-perfect-derives.rs:23:15
+   |
+LL |     let _ = s == s2;
+   |             - ^^ -- S<X, X, X>
+   |             |
+   |             S<X, X, X>
+   |
+note: an implementation of `PartialEq` might be missing for `X`
+  --> $DIR/lack-of-perfect-derives.rs:15:1
+   |
+LL | struct X;
+   | ^^^^^^^^ must implement `PartialEq`
+help: consider annotating `X` with `#[derive(PartialEq)]`
+   |
+LL + #[derive(PartialEq)]
+LL | struct X;
+   |
```

Comment:
This added test is unrelated to the codepath being taken here, but E0369 should eventually use part of the same logic that E0599 does, so that it can provide more suggestions than it does today.